build/root/var/lib/ci-artifacts/entrypoints.d/sro_run_libs.sh

```diff
@@ -0,0 +1,48 @@
+prepare_cluster_for_sro() {
+    toolbox/cluster/capture_environment.sh
+
+    entitle
+
+    if ! toolbox/nfd/has_nfd_labels.sh; then
+        toolbox/nfd/deploy_from_operatorhub.sh
+    fi
+}
+
+validate_sro_deployment() {
+    trap toolbox/special-resource-operator/capture_deployment_state.sh ERR
+
+    toolbox/special-resource-operator/run-e2e-test.sh "${CI_IMAGE_SRO_COMMIT_CI_REPO}" "${CI_IMAGE_SRO_COMMIT_CI_REF}"
+    toolbox/gpu-operator/capture_deployment_state.sh
+}
+
+test_master_branch() {
+    CI_IMAGE_SRO_COMMIT_CI_REPO="${1:-https://github.com/openshift-psap/special-resource-operator.git}"
+    CI_IMAGE_SRO_COMMIT_CI_REF="${2:-master}"
+
+    echo "Using Git repository ${CI_IMAGE_SRO_COMMIT_CI_REPO} with ref ${CI_IMAGE_SRO_COMMIT_CI_REF}"
+
+    prepare_cluster_for_sro
+    toolbox/special-resource-operator/deploy_from_commit.sh "${CI_IMAGE_SRO_COMMIT_CI_REPO}" \
+                                               "${CI_IMAGE_SRO_COMMIT_CI_REF}"
+    validate_sro_deployment
+    exit 0
+}
+
+action="$1"
+shift
+
+set -x
+
+case ${action:-} in
+    "test_master_branch")
+        test_master_branch "$@"
+        ;;
```

Comment:
there is something that I don't like when I read
```
case ${action:-} in
    "test_master_branch")
        test_master_branch "$@"
        ;;
```
it's that it's not clear at all that `test_master_branch` will never return, because it calls `exit 0` or `exit 1`

I was thing about `exec test_master_branch ` but it doesn't work with bash ...
maybe `__test_master_branch` and consider that `__fct` are function never returning,
or `test_master_branch && exit 0 || exit 1`,

I'm not convinced, any better idea? (or @ArangoGutierrez )

Reply:
hum, yes indeed, if the function returns 1 it fails because of `set -e`, so you only need
```
test_master_branch 
exit 0
```